core/common/src/main/java/alluxio/security/LoginUser.java

```diff
@@ -50,18 +50,21 @@ private LoginUser() {} // prevent instantiation
    * @return the login user
    * @throws IOException if login fails
    */
-  public static User get() throws IOException {
+  public static synchronized User get() throws IOException {
```

Comment:
Thanks for the reference! It's pretty interesting to know.

But actually this method is pretty heavily called in security code path. synchronize entire method would probably result in noticeable degrade performance.